### ISSUE TITLE
Get rid of the `event-stream` devDependency

### ISSUE THREAD

**Issue by limonte** [sweetalert2/sweetalert2#1301]
Reasons:

- it's poorly maintained: https://github.com/dominictarr/event-stream/issues/113#issuecomment-423739893
- ⚠️ it's some kind of injection attack: https://github.com/dominictarr/event-stream/issues/116 

We're currently using it in one place, `utils/execute.js`: https://github.com/sweetalert2/sweetalert2/blob/13662349768bfdd98cfb91000e55bf2c3cfc3bb9/utils/execute.js#L2

**Comment by limonte** [comment]
@zenflow you're the author of `utils/execute.js`, do you maybe have any thought what `event-stream` could be replaced with?

**Comment by zenflow** [comment]
@limonte I'll look into this when I have a few free moments. We will likely need to use a couple different modules to replace `es.split()` and `es.merge()`. `.pipe(es.map(...))` can be replaced with `.on('data', ...)`.

For the time being maybe we can just pin the dependency to the exact version that was used when I introduced the dependency. This might even be a sufficient long-term solution since that version has all the features needed and has no bugs affecting us. Thoughts here?

**Comment by gverni** [comment]
Hi @zenflow I'm not familiar with the eventstreams, but let me know if I can be of any help here. If you already have some packages you want me to try, I'd be happy to.

**Comment by zenflow** [comment]
@gverni Wow, thank you very much for offering your help here! 

I don't know of any particular packages to use unfortunately, but I think there should be lots of options. 

You can check [event-stream's documentation](https://github.com/dominictarr/event-stream#eventstream) to see what each of the three functions do, and find a module to replace each. Note that the default for `matcher` in `es.split(matcher)` is `\n`.

Don't hesitate if you have any questions for me, and thanks again!

**Comment by zenflow** [comment]
@gverni Looks like we can probably just use the modules that event-stream itself depends on:

https://github.com/dominictarr/event-stream/blob/9a5c52a712a721cbc76192b5d0940bcc29cbef03/package.json#L10-L17

**Comment by gverni** [comment]
> Looks like we can probably just use the modules that event-stream itself depends on:

@zenflow good point, I didn't think about that. Let me try to workout how `event-stream` uses those dependencies and implement it for our needs. I may be very slow (still learning), so feel free to chime in anytime if we want to get rid of this package quickly.

**Comment by zenflow** [comment]
Looks like the only one of the 3 functions event-stream implements itself (not just exporting an external dependency as-is) is `es.merge()`. For that I think the `merge2` package, which we're already using, should work.

**Comment by zenflow** [comment]
@gverni I opened PR #1303 to remove any urgency on this issue, so you can relax and take your tiime with this.

**Comment by gverni** [comment]
thanks @zenflow. I'll start working on the implementation ASAP!

**Comment by gverni** [comment]
Hi @zenflow @limonte I started looking into this issue. As first step I removed the dependency from `event-stream`, added dependencies to `stream`, `split`, `map-stream` and copy paste the implementation of `merge`. I tested it executing karma and i have not seen any strange behavior. Changes are in 224bf72

Now, the `split`, `map-stream` are from the same author of `event-stream` so I'd be inclined to remove them as well, but I want to better understand how it works. Obviously it's still very much a work in progress, and very far from a clean implementation, but I thought of keeping you updated.
